lib/zipkin-tracer/zipkin_tracer_base.rb

```diff
@@ -23,7 +23,16 @@ def with_new_span(trace_id, name)
 
     def end_span(span)
       span.close
-      if span.annotations.any?{ |ann| ann.value == Annotation::SERVER_SEND }
+      # if in a thread not handling incoming http requests, it will not have Annotation::SERVER_SEND, so the span
+      # will never be flushed and will cause memory leak.
+      # It will have CLIENT_SEND and CLIENT_RECV if the thread sends out http requests, so use CLIENT_RECV as the sign
```

Comment:
Thanks, @jcarres-mdsol . I get the point that extra access to "spans" could be slow. How about we move this code into store_span, and save has_server_recv_span in the span? That method will access spans anyway.
```
has_server_recv_span = spans.any? do |s|
        s.annotations.any? { |ann| ann.value == Annotation::SERVER_RECV }
      end
```

I think the memory leak needs to be fixed, because some tools-like gems (like newrelic) can do faraday calls in new thread, and once you use both newrelic and zipkin, you're leaking memory. The user may not realize he's tracing a http call in a gem and he is leaking memory because of that, even if the user only intends to do local tracing.